### PR TITLE
Rename Dockerfile-connect to Dockerfile-node, update with Rust/wasm-pack

### PR DESCRIPTION
**Pull request by ibeckermayer**
This is required for using this docker file to build webassets once IronRDP gets merged.

`make -C build.assets/ webassets` works for me locally, `make -C build.assets teleterm` fails but it appears to be related to attempting to run an x86 version of something called fpm on my local machine (which is arm64):

```
errorOut=/home/ci/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin/ruby: line 6: /home/ci/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin.real/ruby: cannot execute binary file: Exec format error
/home/ci/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin/ruby: line 6: /home/ci/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin.real/ruby: Success
```

I'm presuming that since this wasn't a problem before, it won't be a problem in actual CI once this is merged.

### CODE REVIEW

File: build.assets/Dockerfile-node [hunk 1]
```diff
@@ -41,12 +45,21 @@ RUN apt-get -y update && \
     # Used during tag builds to build the RPM package of Connect.
     rpm \
     && \
+    # Manually install the wasm-opt binary on ARM64.
```
Comment (camscale) [inline build.assets/Dockerfile-node:48]: this comment does not match the line below - it installs binaryen not wasm-opt. Perhaps say "Manually install the wasm-opt binary from the binaryen package on ARM64" just for clarity. To me, this looks like an old comment someone forgot to update :)

File: build.assets/Makefile [hunk 2]
```diff
@@ -214,23 +214,26 @@ CONNECT_VERSION := $(BUILDBOX_VERSION)-dev
...
-# Builds a Docker buildbox for Linux Connect builds
+# Builds a Docker buildbox for builds that depend on Node.js.
+# This can be used for building webassets as well as Teleport Connect.
 #
-.PHONY:buildbox-connect
-buildbox-connect:
-	if [[ $${DRONE} == "true" ]] && ! docker inspect --type=image $(BUILDBOX_CONNECT) 2>&1 >/dev/null; then docker pull $(BUILDBOX_CONNECT) || true; fi; \
+.PHONY:buildbox-node
+buildbox-node:
+	if [[ $${DRONE} == "true" ]] && ! docker inspect --type=image $(BUILDBOX_NODE) 2>&1 >/dev/null; then docker pull $(BUILDBOX_NODE) || true; fi; \
```
Comment (camscale) [inline build.assets/Makefile:222]: I missed changing this in a recent PR. If you look at the other buildboxes, they start with something like

    $(call PULL_ON_CI,$(BUILDBOX_ARM))

Could you git rid of this line and change it to

    $(call PULL_ON_CI,$(BUILDBOX_NODE))

**Review by camscale — APPROVED** [review]

**Review by jakule — CHANGES_REQUESTED** [review]
I don't think we should be using Connect container to build our webassets. In the past, we used to have one container to build everything, and in the last few months, we started creating containers "per build" instead of "one to rule them all." The main issues that we had with "one for all" were:
1. One change breaks another build that uses the same container
2. It takes a long time to build each container as a lot of dependencies need to be installed.
3. It's hard to update any dependency as some applications may depend on an older version (that was the issue with Node and Connect AFAIR).

Instead, we should create a new container, "node-webassets" or something like that, and install node + wasm. The same as Connect, we can import from the official node container.

@ibeckermayer Let me know if you need any help with that.

**Comment by camscale** [comment]
@zmb3 Would you like to weigh in here? Is this the direction you were talking about with me for building the web assets separately? While we could have different build containers for connect and webassets, my (very limited) understanding was that the tech for both heavily overlap so a common container makes sense, but this tech is outside my areas.

When you asked me about this, did you have in mind using a common buildbox image for both?

**Comment by zmb3** [comment]
I'm totally fine with a common build box for both.

**Review by jakule — APPROVED** [review]
Looks like this will require more work than expected. I don't think the current implementation should be blocked until we have the proper solution.

File: build.assets/Dockerfile-node (hunk 1 above)
Comment (ibeckermayer) [reply build.assets/Dockerfile-node]: f2993a1fb606bf9de154b49e93f98dd4513a11c1

File: build.assets/Makefile (hunk 2 above)
Comment (ibeckermayer) [reply build.assets/Makefile]: f2993a1fb606bf9de154b49e93f98dd4513a11c1

**Pull request merged**